apps/meteor/ee/tests/unit/apps/livechat-enterprise/lib/AutoCloseOnHold.tests.ts

```diff
@@ -0,0 +1,246 @@
+import sinon from 'sinon';
+
+chai.use(chaiDateTime);
+
+const mockAgendaConstructor = sinon.stub();
+const mockAgendaStart = sinon.stub();
+const mockAgendaScheduler = sinon.stub();
+const mockAgendaCancel = sinon.stub();
+const mockAgendaDefine = sinon.stub();
+const mockLivechatCloseRoom = sinon.stub();
+const mockMeteorStartup = sinon.stub();
+const mockLivechatRooms = {
+	findOneById: sinon.stub(),
+};
+const mockUsers = {
+	findOneById: sinon.stub(),
+};
+
+class MockAgendaClass {
+	constructor(opts: Record<string, any>) {
+		mockAgendaConstructor(opts);
+	}
+
+	async start() {
+		return mockAgendaStart();
+	}
+
+	async schedule(...args: any) {
+		return mockAgendaScheduler(...args);
+	}
+
+	async cancel(...args: any) {
+		return mockAgendaCancel(...args);
+	}
+
+	async define(...args: any) {
+		return mockAgendaDefine(...args);
+	}
+}
+
+const infoStub = sinon.stub();
+const debugStub = sinon.stub();
+const mockLogger = {
+	section: sinon.stub().returns({
+		info: infoStub,
+		debug: debugStub,
+	}),
+};
+
+const mocks = {
+	'@rocket.chat/agenda': { Agenda: MockAgendaClass },
+	'meteor/meteor': { Meteor: { startup: mockMeteorStartup } },
+	'meteor/mongo': {
+		MongoInternals: {
+			defaultRemoteCollectionDriver: () => {
+				return {
+					mongo: { client: { db: sinon.stub() } },
+				};
+			},
+		},
+	},
+	'../../../../../app/livechat/server/lib/LivechatTyped': { Livechat: { closeRoom: mockLivechatCloseRoom } },
+	'./logger': { schedulerLogger: mockLogger },
+	'@rocket.chat/models': {
+		LivechatRooms: mockLivechatRooms,
+		Users: mockUsers,
+	},
+};
+
+const { AutoCloseOnHoldSchedulerClass } = proxyquire
+	.noCallThru()
+	.load('../../../../../app/livechat-enterprise/server/lib/AutoCloseOnHoldScheduler', mocks);
+
+describe('AutoCloseOnHoldScheduler', () => {
+	beforeEach(() => {
+		mockMeteorStartup.resetHistory();
+	});
+
+	it('should call logger.section upon instantiating', () => {
+		expect(mockLogger.section.called).to.be.true;
+	});
+
+	describe('init', () => {
+		beforeEach(() => {
+			mockAgendaStart.resetHistory();
+			mockAgendaScheduler.resetHistory();
+		});
+
+		it('should do nothing if scheduler is already running', async () => {
+			const scheduler = new AutoCloseOnHoldSchedulerClass();
+			scheduler.running = true;
+
+			await scheduler.init();
+
+			expect(mockAgendaScheduler.called).to.be.false;
+		});
+
+		it('should succesfully init the scheduler', async () => {
+			const scheduler = new AutoCloseOnHoldSchedulerClass();
+
+			await scheduler.init();
+
+			expect(mockAgendaStart.calledOnce).to.be.true;
+			expect(scheduler.running).to.be.true;
+			expect(infoStub.calledWith('Service started')).to.be.true;
+		});
+	});
+
+	describe('scheduleRoom', () => {
+		beforeEach(() => {
+			mockAgendaCancel.resetHistory();
+			mockAgendaDefine.resetHistory();
+		});
+
+		it('should schedule a room', async () => {
+			const scheduler = new AutoCloseOnHoldSchedulerClass();
+
+			await scheduler.init();
```

Comment:
Can we add a test for when the scheduler has not initialized yet? And maybe add a check for that following the comment below.